gems/smithy-client/spec/smithy-client/codecs/cbor_spec.rb

```diff
@@ -36,18 +44,32 @@ module Codecs
           struct
         end
 
+        let(:union_shape) do
+          shape = Schema::Shapes::UnionShape.new(id: 'union')
+          shape.add_member(:foo_bar, 'fooBar', string_shape, TestUnion::FooBar)
+          shape.type = TestUnion
+          shape
+        end
+
         it 'serializes returns nil when given shape is Prelude::Unit' do
-          expect(subject.serialize('', Schema::Shapes::Prelude::Unit)).to be_nil
+          expect(subject.serialize(Schema::Shapes::Prelude::Unit, '')).to be_nil
         end
 
         it 'deserializes returns an empty hash when given bytes are empty' do
-          expect(subject.deserialize('', string_shape)).to be_empty
+          expect(subject.deserialize(string_shape, '')).to be_empty
         end
 
         it 'serializes and deserializes a complex data' do
           typed = typed_struct.new(s: 'foo', l: %w[foo bar], m: { 'foo' => 'bar'.dup })
-          serialized = subject.serialize(typed, structure_shape)
-          expect(subject.deserialize(serialized, structure_shape)).to eq(typed)
+          serialized = subject.serialize(structure_shape, typed)
+          expect(subject.deserialize(structure_shape, serialized)).to eq(typed)
+        end
+
+        # TODO: Find better home that tests serde lifecycle for union types
+        it 'serializes and deserializes a union data' do
```

Comment:
I made this quick test during for union parsing implementation. 

May be already covered by other specs but I added this to ensure we are correctly serde union data. If it is already covered, could remove.  I think the biggest things that this test helped was SERDE the member names since it can be formatted differently on the model.